src/lms-ds-loader/edfi_lms_ds_loader/loader_facade.py

```diff
@@ -9,29 +9,81 @@
 # looping logic should go into other modules where they can be tested easily.
 
 import logging
+import os
+from typing import Callable, List
 
 from edfi_lms_extractor_lib.helpers.decorators import catch_exceptions
+import pandas as pd
 
 from edfi_lms_ds_loader.helpers.constants import Table
 from edfi_lms_ds_loader.helpers.argparser import MainArguments
 from edfi_lms_ds_loader import migrator
-from edfi_lms_file_utils import file_reader
+from edfi_lms_file_utils import file_reader, file_repository
+from edfi_lms_file_utils.constants import Resources
 from edfi_lms_ds_loader import df_to_db
 from edfi_lms_ds_loader.mssql_lms_operations import MssqlLmsOperations
 
 logger = logging.getLogger(__name__)
 
 
+def _get_unprocessed_file_paths(
+    db_adapter: MssqlLmsOperations,
+    resource_name: str,
+    get_all_files_callback: Callable,
+):
+    processed_files = db_adapter.get_processed_files(resource_name)
+    all_files: List[str] = get_all_files_callback()
+    all_files_df = pd.DataFrame(all_files, columns=["FullPath"])
+
+    unprocessed_files = pd.concat([processed_files, all_files_df]).drop_duplicates(
+        keep=False
+    )
```

Comment:
This seems really weird. Why would `unprocessed_files` be created by combining the list of _already processed files_ with the list of all files available? I think I see the answer because of the dropping of duplicates. But it seems really weird reading this way, doesn't it?

https://stackoverflow.com/a/61143201/30384 offers another approach. I'm not convinced that it is any better though!

Seems like these are just lists. Perhaps it would make more sense not to use DataFrames at all.

```python
processed_files: Set[str] = db_adapter.get_processed_files(resource_name)
all_files: Set[str = get_all_files_callback()

return all_files - processed_files
```

Reply:
set does not support that operation, will use a list instead for the return type of get_all_files_callback